addons/payment/static/src/js/post_processing.js

```diff
@@ -76,14 +76,6 @@ odoo.define('payment.post_processing', function (require) {
                 'tx_error': [],
             };
 
-            if (display_values_list.length > 0) {
-                // In almost every cases there will be a single transaction to display. If there are
-                // more than one transaction, the last one will most likely be the one that was
-                // confirmed. We use this one to redirect the user to the final page.
-                window.location = display_values_list[0].landing_route;
-                return;
-            }
-
```

Comment:
if I keep this the same issue will rise 
no matter the number/state of the transactions we will always be redirected to the first one

Reply:
I understand your concern but I don't think that the same issue will rise here. This code was introduced in 15+ versions during the payment apocalypse and changed the behavior of the payment processing page. This was not intended but we don't want to change the behavior in stable versions, especially since it's working well. A task in 16+ needed a change of behavior anyway so that's why this block of code doesn't appear in newer versions.

Reply:
@chevalierv it's done thanks ;)